exporter/exporterhelper/internal/persistent_storage_test.go

```diff
@@ -32,6 +32,10 @@ import (
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
+var (
+	mockClientCloseCounting = uint64(0)
```

Comment:
Why global instead of a member in the mockStorageClient

Reply:
If we put it inside `mockStorageClient`, we have to cast the type and extract the value. I just thought that would be uglier.

If you have strong preference to leave it in `mockStorageClient` I can make the change.

Reply:
On second thought the counter won't work if there are multiple clients created. I will move it inside.